src/Donation/DonationRequestUtils.php

```diff
@@ -115,15 +115,15 @@ public function createCallbackStatus(Donation $donation): array
 
     private function hydrateFromRetryPayload(DonationRequest $request, string $payload): DonationRequest
     {
-        $data = array_filter(json_decode($payload, true));
+        $data = array_filter(json_decode(urldecode($payload), true));
```

Comment:
The payload is not checked to be valid JSON, this could easily fail. We should handle the case IMO.